### PR TITLE
New validate BC105

### PR DESCRIPTION
**Pull request by anas-yousef**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9803

## Description
Added **BC105** validation to new validation manager.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/BC_validators/BC105_id_changed.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    ParsingRule,
+    ModelingRule,
+    XSIAMDashboard,
+    Trigger,
+    XSIAMReport,
+    IncidentField,
+    IndicatorField,
+    AssetsModelingRule,
+    LayoutRule,
+]
```
Comment (anas-yousef) [inline demisto_sdk/commands/validate/validators/BC_validators/BC105_id_changed.py:73]: @YuvHayun Do all content items have the field **object_id** configured?

File: .changelog/4073.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added support for validation BC105 to the new validation manager.
+  type: internal
+pr_number: 4073
```
Comment (YuvHayun) [inline .changelog/4073.yml:4]: Remove, we don't announce on such changes.

File: demisto_sdk/commands/validate/validators/BC_validators/BC105_id_changed.py (hunk 1 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validators/BC_validators/BC105_id_changed.py:73]: Yap.

File: demisto_sdk/commands/validate/validators/BC_validators/BC105_id_changed.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+
+class IdChangedValidator(BaseValidator[ContentTypes]):
+    error_code = "BC105"
+    description = "Validate that the ID of the content item was not changed."
+    error_message = "ID of content item was changed from {0} to {1}, please undo."
+    fix_message = "Changing ID back to {0}."
+    related_field = "id"
+    expected_git_statuses = [GitStatuses.MODIFIED]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BC_validators/BC105_id_changed.py:82]: Let's add Renamed as well.

**Review by YuvHayun — APPROVED** [review]
Looks good, minor comments.

**Review by JasBeilin — APPROVED** [review]

**Pull request merged**